src/charm.py

```diff
@@ -401,11 +400,11 @@ def set_unit_status(
         if (
             isinstance(status, ops.ActiveStatus)
             and refresh is not None
-            and refresh.unit_status_lower_priority()
+            and (refresh_status := refresh.unit_status_lower_priority())
```

Comment:
Reusing the return status, otherwise the type checker is not sure if it's None or a value.